code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -525,6 +525,17 @@
 		desc = "Your standard drinking glass."
 		return
 
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/attackby(obj/item/I, mob/user, params)
+	if(!user.canUseTopic(src, be_close=TRUE))
+		return
+	if(istype(I, /obj/item/weapon/pen))
+		var/txt = stripped_input(user, "What would you like to name this drink?", "Mixology", "", 30)
+		if(txt)
```

Comment:
That does not prevent anyone from opening the `stripped_input` and keeping it open.